### PR TITLE
add controller to sync aws resource tags

### PR DESCRIPTION
**Pull request by staebler**
Add aws_resource_tags controller. The controller will sync AWS resource tags from the spec of the infrastructure resource to the status of the resource.

/cc @deads2k

### CODE REVIEW

**Comment by staebler** [comment]
/hold

This is using a fork of openshift/api until https://github.com/openshift/api/pull/905 merges.

File: pkg/operator/aws_resource_tags/controller.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+		}
+	}
+
+	var currentResourceTags []configv1.AWSResourceTag
+	if currentInfra.Status.PlatformStatus != nil &&
+		currentInfra.Status.PlatformStatus.AWS != nil {
+		currentResourceTags = currentInfra.Status.PlatformStatus.AWS.ResourceTags
+	}
+
+	if len(desiredResourceTags) == 0 && len(currentResourceTags) == 0 {
```
Comment (deads2k) [inline pkg/operator/aws_resource_tags/controller.go:91]: seems like this case is handled by the deepequal on the next line.  is it not?

File: pkg/operator/aws_resource_tags/controller.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+	if currentInfra.Status.PlatformStatus == nil {
+		currentInfra.Status.PlatformStatus = &configv1.PlatformStatus{}
+	}
+	if currentInfra.Status.PlatformStatus.AWS == nil {
+		currentInfra.Status.PlatformStatus.AWS = &configv1.AWSPlatformStatus{}
+	}
+	currentInfra.Status.PlatformStatus.AWS.ResourceTags = desiredResourceTags
+
+	_, err = c.infraClient.UpdateStatus(ctx, currentInfra, metav1.UpdateOptions{})
```
Comment (deads2k) [inline pkg/operator/aws_resource_tags/controller.go:107]: I think an event on successful update is also warranted.

File: pkg/operator/aws_resource_tags/controller.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+		return nil
+	}
+
+	if currentInfra.Status.PlatformStatus == nil {
+		currentInfra.Status.PlatformStatus = &configv1.PlatformStatus{}
+	}
+	if currentInfra.Status.PlatformStatus.AWS == nil {
+		currentInfra.Status.PlatformStatus.AWS = &configv1.AWSPlatformStatus{}
+	}
+	currentInfra.Status.PlatformStatus.AWS.ResourceTags = desiredResourceTags
```
Comment (deads2k) [inline pkg/operator/aws_resource_tags/controller.go:105]: Unless order matters, please sort the status resourcetags.

**Comment by deads2k** [comment]
minor comments, but I think the API and additional validation make sense and provide clear indications of what can and cannot be done.

**Pull request closed without merging**